### PR TITLE
Update Isar layers, improve openssl rebuild

### PR DESCRIPTION
**Pull request by jan-kiszka**
Work in progress: more testing needed, and one more catch-up with isar-cip-core is missing.

@adriaan42

### CODE REVIEW

File: recipes-core/images/iot2050-image-example.bb [hunk 1]
```diff
@@ -13,7 +13,7 @@ require recipes-core/images/iot2050-package-selections.inc
 
 DESCRIPTION = "IOT2050 Debian Example Image"
 
-DEPENDS += "openssl"
+DEPENDS_class-sdk += "openssl-native"
```
Comment (adriaan42) [inline recipes-core/images/iot2050-image-example.bb:16]: Correct syntax would be `DEPENDS:class-sdk`, but in this case you shouldn't need this at all. `openssl` is in `ISAR_REBUILD_ESSENTIAL_PKGS`, and all variants should be built before any other packages (`do_prepare_build` depends on them) or images (`do_rootfs_install` depends on them).

File: recipes-core/images/iot2050-image-swu-example.bb [hunk 2]
```diff
@@ -13,11 +13,11 @@ require recipes-core/images/swupdate.inc
 
 inherit image_uuid
 
-IMAGE_TYPEDEP_wic = "squashfs"
-IMAGE_TYPEDEP_wic_secureboot = "verity"
+IMAGE_TYPEDEP:wic = "squashfs"
+IMAGE_TYPEDEP:wic:secureboot = "verity"
```
Comment (adriaan42) [inline recipes-core/images/iot2050-image-swu-example.bb:17]: This looks strange... should this be `IMAGE_TYPEDEP:secureboot`?

File: recipes-initramfs/iot2050-initramfs/iot2050-initramfs_0.1.bb [hunk 3]
```diff
@@ -11,8 +11,8 @@
...
 
 INITRAMFS_INSTALL += " \
-    initramfs-etc-overlay-hook \
+    initramfs-overlay-hook \
     initramfs-abrootfs-hook \
     "
-INITRAMFS_INSTALL_append_secureboot += " initramfs-verity-hook"
-INITRAMFS_INSTALL_remove_secureboot += " initramfs-abrootfs-hook"
+INITRAMFS_INSTALL:append:secureboot = " initramfs-verity-hook"
+INITRAMFS_INSTALL:remove:secureboot = " initramfs-abrootfs-hook"
```
Comment (adriaan42) [inline recipes-initramfs/iot2050-initramfs/iot2050-initramfs_0.1.bb:18]: `<nitpick>`with `:remove` you don't need the extra space`</nitpick>`

File: recipes-core/images/iot2050-image-example.bb (hunk 1 above)
Comment (jan-kiszka) [reply recipes-core/images/iot2050-image-example.bb:16]: Ah, left-over. Will drop.

File: recipes-core/images/iot2050-image-swu-example.bb (hunk 2 above)
Comment (jan-kiszka) [reply recipes-core/images/iot2050-image-swu-example.bb:17]: It's actually a conditional: If we build a non-secure image, we use plain squashfs. If secure is on, that `secureboot` override is set, and we wrap dm-verity around the squashfs. But all that is prior to Felix' recent isar-cip-core image type changes. Those are my major todo, and maybe they will change this as well.

File: conf/distro/iot2050-debian.conf [hunk 4]
```diff
@@ -35,24 +35,24 @@ PACKAGES_SELECTION ?= "packages-mainline"
...
 
 SDK_FORMATS = "tar.xz docker-archive"
 
 # used for SWUpdate boot
-IMAGE_CLASSES += "squashfs verity"
 ABROOTFS_IMAGE_RECIPE ?= "iot2050-image-swu-example"
 VERITY_IMAGE_RECIPE ?= "iot2050-image-swu-example"
 INITRAMFS_RECIPE ?= "iot2050-initramfs"
+ABROOTFS_PART_UUID_A ?= "fedcba98-7654-3210-cafe-5e0710000001"
+ABROOTFS_PART_UUID_B ?= "fedcba98-7654-3210-cafe-5e0710000002"
```
Comment (fmoessbauer) [inline conf/distro/iot2050-debian.conf:56]: This should be machine specific, not distro specific.

Comment (jan-kiszka) [reply conf/distro/iot2050-debian.conf:56]: This is not machine specific, it's wic-file specific. So, this does not matter here.

File: wic/iot2050-swu.wks.in [hunk 5]
```diff
@@ -10,8 +10,8 @@
 
 include ebg-sysparts.inc
 
-part --source rawcopy --sourceparams "file=${IMAGE_FULLNAME}.squashfs" --align 1024 --fixed-size 4G --uuid "fedcba98-7654-3210-cafe-5e0710000001"
-part --source rawcopy --sourceparams "file=${IMAGE_FULLNAME}.squashfs" --align 1024 --fixed-size 4G --uuid "fedcba98-7654-3210-cafe-5e0710000002"
+part --source rawcopy --sourceparams "file=${IMAGE_FULLNAME}.squashfs" --align 1024 --fixed-size 4G --uuid "${ABROOTFS_PART_UUID_A}"
+part --source rawcopy --sourceparams "file=${IMAGE_FULLNAME}.squashfs" --align 1024 --fixed-size 4G --uuid "${ABROOTFS_PART_UUID_B}"
```
Comment (BaochengSu) [inline wic/iot2050-swu.wks.in:14]: The `squashfs` and `verity` is also used as the `IMAGE_TYPEDEP` in recipes-core/images/iot2050-image-swu-example.bb, is it possible to define some variables to hold them?

File: recipes-app/iot2050-firmware-update/iot2050-firmware-update_0.4.bb
Comment (BaochengSu) [inline recipes-app/iot2050-firmware-update/iot2050-firmware-update_0.4.bb:1]: 2020-2023

File: recipes-bsp/optee-os/optee-os-iot2050_3.19.0.bb
Comment (BaochengSu) [inline recipes-bsp/optee-os/optee-os-iot2050_3.19.0.bb:1]: 2020-2023

File: recipes-bsp/secure-boot-otp-provisioning/secure-boot-otp-provisioning_0.1.bb
Comment (BaochengSu) [inline recipes-bsp/secure-boot-otp-provisioning/secure-boot-otp-provisioning_0.1.bb:1]: 2022-2023

File: recipes-core/iot2050-firmware/iot2050-firmware_0.1.bb
Comment (BaochengSu) [inline recipes-core/iot2050-firmware/iot2050-firmware_0.1.bb:1]: 2021-2023

File: wic/iot2050-swu-secure.wks.in
Comment (BaochengSu) [inline wic/iot2050-swu-secure.wks.in:1]: 2021-2023

File: wic/iot2050-swu.wks.in
Comment (BaochengSu) [inline wic/iot2050-swu.wks.in:1]: 2021-2023

File: classes/npm.bbclass
Comment (BaochengSu) [inline classes/npm.bbclass:1]: Eventually `buildchroot` will be dropped?

File: recipes-bsp/k3-rti-wdt/k3-rti-wdt_1.1.bb
Comment (BaochengSu) [inline recipes-bsp/k3-rti-wdt/k3-rti-wdt_1.1.bb:1]: # Copyright (c) Siemens AG, 2021-2023

File: wic/iot2050-swu.wks.in (hunk 5 above)
Comment (jan-kiszka) [reply wic/iot2050-swu.wks.in:14]: I can't follow yet: What exactly should be shared via variables?

File: classes/npm.bbclass
Comment (jan-kiszka) [reply classes/npm.bbclass:1]: Yes, once the imager is based on an own chroot. There are patches for that already on the Isar list.

File: recipes-bsp/k3-rti-wdt/k3-rti-wdt_1.1.bb
Comment (jan-kiszka) [reply recipes-bsp/k3-rti-wdt/k3-rti-wdt_1.1.bb:1]: Done already in another patch.

File: wic/iot2050-swu.wks.in
Comment (jan-kiszka) [reply wic/iot2050-swu.wks.in:1]: done

File: wic/iot2050-swu.wks.in (hunk 5 above)
Comment (BaochengSu) [reply wic/iot2050-swu.wks.in:14]: make `SWU_ROOTFS_TYPE` a global variable and use it here:

```suggestion
part --source rawcopy --sourceparams "file=${IMAGE_FULLNAME}.${SWU_ROOTFS_TYPE}" --align 1024 --fixed-size 4G --uuid "${ABROOTFS_PART_UUID_A}"
part --source rawcopy --sourceparams "file=${IMAGE_FULLNAME}.${SWU_ROOTFS_TYPE}" --align 1024 --fixed-size 4G --uuid "${ABROOTFS_PART_UUID_B}"
```

Then you can combine some of the `iot2050-swu.wks.in` and `iot2050-swu-secure.wks.in`.

Comment (jan-kiszka) [reply wic/iot2050-swu.wks.in:14]: Nice idea. Unfortunately, variable resolution in wic does not work on include files, and that would be the only way to share those bits between the two wks files.

Comment (jan-kiszka) [reply wic/iot2050-swu.wks.in:14]: But I just noticed that we can make B smaller by leaving it empty, just like cip-core does now.

**Review by BaochengSu — APPROVED** [review]

**Comment by BaochengSu** [comment]
@fmoessbauer @adriaan42 Any new comments? If not, I will merge it soon.

**Review by adriaan42 — APPROVED** [review]

**Pull request merged**